beacon-chain/core/helpers/validators.go

```diff
@@ -175,20 +175,30 @@ func ValidatorChurnLimit(activeValidatorCount uint64) (uint64, error) {
 //    return compute_proposer_index(state, indices, seed)
 func BeaconProposerIndex(state *stateTrie.BeaconState) (uint64, error) {
 	e := CurrentEpoch(state)
-
-	seed, err := Seed(state, e, params.BeaconConfig().DomainBeaconAttester)
-	if err != nil {
-		return 0, errors.Wrap(err, "could not generate seed")
-	}
-	proposerIndices, err := committeeCache.ProposerIndices(seed)
-	if err != nil {
-		return 0, errors.Wrap(err, "could not interface with committee cache")
-	}
-	if proposerIndices != nil {
-		return proposerIndices[state.Slot()%params.BeaconConfig().SlotsPerEpoch], nil
+	// The cache uses the block root of the previous epoch's last slot as key. (e.g. Starting epoch 1, slot 32, the key would be block root at slot 31)
+	// For simplicity, the node will skip caching of genesis epoch.
+	if e > params.BeaconConfig().GenesisEpoch {
+		s, err := EndSlot(PrevEpoch(state))
```

Comment:
can we just make `EndSlot(PrevEpoch)` a helper of its own ? dont feel too strongly about it